modules/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobContainerRetriesTests.java

```diff
@@ -173,6 +208,29 @@ StorageOptions createStorageOptions(
         return new GoogleCloudStorageBlobContainer(randomBoolean() ? BlobPath.EMPTY : BlobPath.EMPTY.add("foo"), blobStore);
     }
 
+    public void testShouldRetryOnNetworkOutage() {
+        final int maxRetries = randomIntBetween(3, 5);
+
+        if (randomBoolean()) {
+            logger.info("Failing due to connection refused");
+            endpointUrlOverride = "http://127.0.0.1:"
+                + randomValueOtherThan(httpServer.getAddress().getPort(), () -> randomIntBetween(49152, 65535));
+        } else {
+            logger.info("Failing due to unknown domain");
+            // https://www.rfc-editor.org/rfc/rfc2606.html#page-2
+            endpointUrlOverride = "http://unresolvable.invalid";
+        }
+
+        BlobContainer blobContainer = createBlobContainer(maxRetries, null, null, null, null);
+        try {
+            blobContainer.listBlobs(randomPurpose());
+            fail("Should have thrown an exception");
```

Comment:
I'd prefer `expectThrows` over doing this manually.

Reply:
Addressed in 8b2e376dd090dbe5d22c91cd289b174c8c60f635